test/e2e_node/container.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+//One pod one container
+type ConformanceContainer struct {
+	Container api.Container
+	Client    *client.Client
+	Status    string
+}
+
+func NewConformanceContainer(cl *client.Client, container api.Container) ConformanceContainer {
+	return ConformanceContainer{
+		Container: container,
+		Client:    cl,
+	}
+}
+
+func (cc *ConformanceContainer) Create() error {
+	pod := &api.Pod{
+		ObjectMeta: api.ObjectMeta{
+			//Same with the container name
+			Name:      cc.Container.Name,
+			Namespace: api.NamespaceDefault,
+		},
+		Spec: api.PodSpec{
+			NodeName:      *nodeName,
+			RestartPolicy: api.RestartPolicyNever,
+			Containers: []api.Container{
+				cc.Container,
+			},
+		},
+	}
+
+	_, err := cc.Client.Pods(api.NamespaceDefault).Create(pod)
+	return err
+}
+
+func (cc *ConformanceContainer) IsStarted() (bool, error) {
+	_, cstatus, err := cc.Get()
+	if err != nil {
+		return false, errors.New("Failed to get container")
+	}
+	if cstatus.State.Running != nil || cstatus.State.Terminated != nil {
+		return true, nil
+	} else {
+		//Default to waiting
+		return false, errors.New("Still waiting")
+	}
+}
+
+//Same with 'delete' currently
+func (cc *ConformanceContainer) Stop() error {
+	return cc.Client.Pods(api.NamespaceDefault).Delete(cc.Container.Name, &api.DeleteOptions{})
+}
+
+func (cc *ConformanceContainer) Delete() error {
+	return cc.Client.Pods(api.NamespaceDefault).Delete(cc.Container.Name, &api.DeleteOptions{})
+}
+
+func (cc *ConformanceContainer) Get() (container api.Container, cstatus api.ContainerStatus, err error) {
+	pod, err := cc.Client.Pods(api.NamespaceDefault).Get(cc.Container.Name)
+	if err != nil {
+		return container, cstatus, err
+	}
+
+	containers := pod.Spec.Containers
+	if containers == nil || len(containers) != 1 {
+		return container, cstatus, errors.New("Failed to get container")
+	}
+
+	cstatuss := pod.Status.ContainerStatuses
+	if cstatuss == nil || len(cstatuss) != 1 {
+		return container, cstatus, errors.New("Failed to get container status")
+	}
+	return containers[0], cstatuss[0], nil
+}
+
+func (cc *ConformanceContainer) Equal(container api.Container, cstatus api.ContainerStatus) error {
```

Comment:
I think the not Equal condition might be better represented as a bool.  We could either log the message or return it as a second string argument.